src/main/java/com/unzer/payment/paymenttypes/Applepay.java

```diff
@@ -33,116 +33,134 @@
-		((Applepay) applepay).setTransactionAmount(((JsonApplepayResponse) jsonApplePay).getTransactionAmount());
-		((Applepay) applepay).setRecurring(((JsonApplepayResponse) jsonApplePay).getRecurring());
-		GeoLocation tempGeoLocation = new GeoLocation(((JsonApplepayResponse) jsonApplePay).getGeoLocation().getClientIp(), ((JsonApplepayResponse) jsonApplePay).getGeoLocation().getCountryIsoA2());
-		((Applepay) applepay).setGeoLocation(tempGeoLocation);
-		return applepay;
-	}
-
-	public Authorization authorize(BigDecimal amount, Currency currency, URL returnUrl) throws HttpCommunicationException {
-		return authorize(amount, currency, returnUrl, null);
-	}
-	public Authorization authorize(BigDecimal amount, Currency currency, URL returnUrl, Customer customer) throws HttpCommunicationException {
-		return getUnzer().authorize(amount, currency, this, returnUrl, customer);
-	}
-
-	public Charge charge(BigDecimal amount, Currency currency, URL returnUrl) throws HttpCommunicationException {
-		return getUnzer().charge(amount, currency, this, returnUrl);
-	}
-	public Charge charge(BigDecimal amount, Currency currency, URL returnUrl, Customer customer) throws HttpCommunicationException {
-		return getUnzer().charge(amount, currency, this, returnUrl, customer);
-	}
-
-	public String getVersion() {
-		return version;
-	}
-
-	public void setVersion(String version) {
-		this.version = version;
-	}
-
-	public String getData() {
-		return data;
-	}
-
-	public void setData(String data) {
-		this.data = data;
-	}
-	
-	public String getSignature() {
-		return signature;
-	}
-
-	public void setSignature(String signature) {
-		this.signature = signature;
-	}
-
-	public ApplepayHeader getHeader() {
-		return header;
-	}
-
-	public void setHeader(ApplepayHeader header) {
-		this.header = header;
-	}
-
-	public String getCurrencyCode() {
-		return currencyCode;
-	}
-
-	public void setCurrencyCode(String currencyCode) {
-		this.currencyCode = currencyCode;
-	}
-
-	public BigDecimal getTransactionAmount() {
-		return transactionAmount;
-	}
-
-	public void setTransactionAmount(BigDecimal transactionAmount) {
-		this.transactionAmount = transactionAmount;
-	}
-
-	public String getNumber() {
-		return number;
-	}
-
-	public void setNumber(String number) {
-		this.number = number;
-	}
-
-	public String getExpiryDate() {
-		return expiryDate;
-	}
-
-	public void setExpiryDate(String expiryDate) {
-		this.expiryDate = expiryDate;
-	}
-
+    private String version;
+    private String data;
+    private String signature;
+    private ApplepayHeader header;
+
+    private String method;
+    private String applicationPrimaryAccountNumber;
+    private String applicationExpirationDate;
+    private String currencyCode;
+    private BigDecimal transactionAmount;
+
+    public Applepay() {
+    }
+
+    public Applepay(String version, String data, String signature) {
+        this.version = version;
+        this.data = data;
+        this.signature = signature;
+    }
```

Comment:
I did not notice it before, but shouldn't constructor already contain header yet, since it is mandatory